StreamVideoTests/Utils/Sorting_Tests.swift

```diff
@@ -4,182 +4,741 @@
-    }
-    
-    func testPinnedMultipleLocalAndRemote() {
-        // Given
-        let participant1 = makeCallParticipant(id: "1")
-        let participant2 = makeCallParticipant(id: "2", pin: PinInfo(isLocal: true, pinnedAt: Date()))
-        let participant3 = makeCallParticipant(id: "3", pin: PinInfo(isLocal: false, pinnedAt: Date()))
-        let participant4 = makeCallParticipant(id: "4")
-        let participant5 = makeCallParticipant(id: "5", pin: PinInfo(isLocal: true, pinnedAt: Date().addingTimeInterval(-10)))
-        let participant6 = makeCallParticipant(id: "6", pin: PinInfo(isLocal: false, pinnedAt: Date().addingTimeInterval(-10)))
-        let participants = [participant1, participant2, participant3, participant4, participant5, participant6]
-        
-        // When
-        let sorted = participants.sorted(using: [pinned, userId])
-        
-        // Then
-        XCTAssertEqual(sorted.map(\.id), ["3", "6", "2", "5", "4", "1"])
-    }
-    
-    //MARK: - private
-    
-    private func makeCallParticipant(
-        id: String,
-        name: String = "",
-        roles: [String] = [],
-        hasVideo: Bool = false,
-        hasAudio: Bool = false,
-        isScreenSharing: Bool = false,
-        isSpeaking: Bool = false,
-        isDominantSpeaker: Bool = false,
-        pin: PinInfo? = nil
-    ) -> CallParticipant {
-        mockResponseBuilder.makeCallParticipant(
-            id: id,
-            name: name,
-            roles: roles,
-            hasVideo: hasVideo,
-            hasAudio: hasAudio,
-            isScreenSharing: isScreenSharing,
-            isSpeaking: isSpeaking,
-            isDominantSpeaker: isDominantSpeaker,
-            pin: pin
-        )
-    }
-    
+
+    // MARK: - pinned
+
+    /// Test with only pinned participants.
+    func test_pinned_allPinned() {
+        assertSort(
+            [
+                .dummy(pin: PinInfo(isLocal: true, pinnedAt: Date(timeIntervalSince1970: 100))),
+                .dummy(pin: PinInfo(isLocal: true, pinnedAt: Date(timeIntervalSince1970: 200)))
+            ],
+            comparator: pinned,
+            expectedTransformer: { [$0[1], $0[0]] }
+        )
+    }
+
+    /// Test with only unpinned participants.
+    func test_pinned_noPinned() {
+        assertSort(
+            [
+                .dummy(),
+                .dummy()
+            ],
+            comparator: pinned,
+            expectedTransformer: { [$0[0], $0[1]] }
+        )
+    }
+
+    /// Test with a mix of pinned and unpinned participants.
+    func test_pinned_mixedPinned() {
+        assertSort(
+            [
+                .dummy(),
+                .dummy(pin: PinInfo(isLocal: true, pinnedAt: Date(timeIntervalSince1970: 100)))
+            ],
+            comparator: pinned,
+            expectedTransformer: { [$0[1], $0[0]] }
+        )
+    }
+
+    /// Test with pinned participants having different pinnedAt dates.
+    func test_pinned_differentPinnedDates() {
```

Comment:
I don't see a test with local and remote pin. That would've caught the issue with the wrong comparator.

Reply:
I just added one.